control-plane/src/health.rs

```diff
@@ -67,38 +63,31 @@ async fn run_ecs_health_check_service(
     let combined_log_json = serde_json::to_string(&combined_log).unwrap();
 
     Response::builder()
-        .status(status_to_return.status_code())
+        .status(status_to_return)
         .header("Content-Type", "application/json")
         .body(Body::from(combined_log_json))
         .map_err(ServerError::from)
 }
 
-macro_rules! unwrap_or_err {
-    ($result:expr) => {
-        match $result {
-            Ok(ok) => ok,
-            Err(error) => {
-                return HealthCheckLog::new(HealthCheckStatus::Err, Some(error.to_string()))
-            }
-        }
-    };
-}
+type EcsHealthCheckResult = Result<DataPlaneState, ServerError>;
+
+async fn health_check_data_plane() -> EcsHealthCheckResult {
```

Comment:
Is this not a breaking change? Existing Data Plane versions will continue returning a HealthCheckLog, but the Control Plane is going to deserialize it as a `DataPlaneState` instance